packages/devtools_app/lib/src/utils.dart

```diff
@@ -389,7 +389,8 @@ class TimeRange {
 
   bool overlaps(TimeRange t) {
     return (t.start >= start && t.start <= end) ||
```

Comment:
you don't need three clauses for this. 
What you want is
```dart
t.end > start && end > t.start
```
Because the two intervals don't intersect if the start of one is after the end of the other. Additionally, when both clauses hold the rectangles have to intersect because [your proof goes here].
I also think you probably want to not treat intersection at a single point as an overlap as you can still render the ui correctly if the two time ranges bump against each other.